src/PrestaShopBundle/Bridge/AdminController/ControllerConfiguration.php

```diff
@@ -223,21 +220,41 @@ class ControllerConfiguration
      * @param Employee $user the current user
      */
     public function __construct(
-        Employee $user
+        Employee $user,
+        int $tabId,
+        string $objectModelClassName,
+        string $legacyControllerName,
+        string $tableName,
+        string $templateFolder
     ) {
         $this->user = $user;
+        $this->tabId = $tabId;
+        $this->objectModelClassName = $objectModelClassName;
+        $this->legacyControllerName = $legacyControllerName;
+        $this->tableName = $tableName;
+        $this->templateFolder = $templateFolder;
     }
 
     /**
      * Adds toolbar action to the page
      *
-     * @param HeaderToolbarAction $action
+     * @param string $label
+     * @param array<string, mixed> $config
      *
      * @return ControllerConfiguration
      */
-    public function addToolbarAction(HeaderToolbarAction $action): ControllerConfiguration
+    public function addHeaderToolbarAction(string $label, array $config): self
     {
-        $this->pageHeaderToolbarButtons[$action->getLabel()] = $action->getConfig();
+        $optionsResolver = new OptionsResolver();
```

Comment:
The resolver could be created in the constructor to avoid duplicating it on each call (same idea for the other methods relaying on a resolver)

Reply:
I think resolver is not supposed to be used that way. One resolver = one definition of options. So if I only use one resolver the options would mix up. See screenshot of debugger 1. I called testResolver - it defines "test" option. 2.addHeaderToolbarAction is called and as you see there is already "test" option defined, which is not acceptable for headerToolbar options. (its a leftover from previous calls). So for every stack of options, there must be a dedicated new options resolver.